devices/Bmm150/Bmm150.cs

```diff
@@ -0,0 +1,359 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Device.Model;
+using System.Diagnostics;
+using System.IO;
+using System.Numerics;
+using System.Threading;
+
+namespace Iot.Device.Magnetometer
+{
+    /// <summary>
+    /// Bmm150 class implementing a magnetometer
+    /// </summary>
+    [Interface("Bmm150 class implementing a magnetometer")]
+    public sealed class Bmm150 : IDisposable
+    {
+        /// <summary>
+        /// I2c device comm channel
+        /// </summary>
+        private I2cDevice _i2cDevice;
+
+        /// <summary>
+        /// Bmm150 device interface
+        /// </summary>
+        private Bmm150I2cBase _Bmm150Interface;
+
+        /// <summary>
+        /// Bmm150 Trim extended register data
+        /// </summary>
+        private Bmm150TrimRegisterData _trimData;
+
+        /// <summary>
+        /// Magnetometer calibration compensation vector
+        /// </summary>
+        private Vector3 _calibrationCompensation;
```

Comment:
Why not having the calibration data exposed? It's the case for the AK8963 for example. If anyone wants to adjust them for any reason or not go thru the proposed calibration process but with they own, we should let them place their own.

Reply:
done